src/camelsnake/Converter.java

```diff
@@ -0,0 +1,89 @@
+package camelsnake;
+
+public class Converter {
+
+    private enum Type {
+        UPPER,
+        LOWER,
+        DIGIT,
+        DASH,
+        UNDERSCORE,
+        OTHER
+    }
+
+    static private Type getCharType(char ch) {
+        if (Character.isUpperCase(ch)) {
+            return Type.UPPER;
+        } else if (Character.isLowerCase(ch)) {
```

Comment:
I assume lower case is most common.  If you want to get every possible cycle, you could order it as the first if statement check.